app/src/main/java/com/example/whatsapp_clone/Views/Fragments/RegisterFragment.java

```diff
@@ -1,38 +1,203 @@
 package com.example.whatsapp_clone.Views.Fragments;
 
+import static android.app.Activity.RESULT_OK;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.lifecycle.ViewModelProvider;
 
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
 import android.os.Bundle;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
 
+import android.provider.MediaStore;
+import android.util.Base64;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
 
 import com.example.whatsapp_clone.R;
+import com.example.whatsapp_clone.ValidationTester;
+import com.example.whatsapp_clone.databinding.FragmentRegisterBinding;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
 
 public class RegisterFragment extends Fragment {
 
+    private FragmentRegisterBinding binding;
+    private Button registerBtn;
     private RegisterViewModel mViewModel;
+    private EditText userEmailInput;
+    private EditText userDisplayNameInput;
+    private EditText userPasswordInput;
+    private EditText userPasswordConfInput;
+
+    private static final int REQUEST_IMAGE_PICK = 1;
+    private ImageView profileImageView;
+    private Button uploadPictureBtn;
+
+    private String base64ProfilePic;
+    private ActivityResultLauncher<Intent> imagePickerLauncher;
+
+
 
     public static RegisterFragment newInstance() {
         return new RegisterFragment();
     }
 
+
     @Override
```

Comment:
All UI Events such as clicks should be moved to onViewCreated
this method should only initialize the binding and return the rootView